include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1277,6 +1281,38 @@ struct OrtApi {
      */
   ORT_API2_STATUS(KernelInfoGetAttributeArray_int64, _In_ const OrtKernelInfo* info, _In_ const char* name,
                   _Out_ int64_t* out, _Inout_ size_t* size);
+
+  /**
+  * Use this API to create the configuration of an arena that can eventually be used to define
+  * an arena based allocator's behavior
+  * \param arena_config_keys - keys to configure the arena
+  * \param arena_config_values - values to configure the arena
+  * \param num_keys - number of keys passed in
+  * Supported keys are (See docs/C_API.md for details on what the following parameters mean and how to choose these values.):
+  * "max_mem": Maximum memory that can be allocated by the arena based allocator. 
+     Use 0 for ORT to pick the best value. Default is 0.
+  * "arena_extend_strategy": 0 = kNextPowerOfTwo, 1 = kSameAsRequested. 
+     Use -1 to allow ORT to choose the default.
+  * "initial_chunk_size_bytes": Size of the first allocation in the arena. 
+     Only relevant if arena strategy is `kNextPowerOfTwo`. Use -1 to allow ORT to choose the default.
+  * "max_dead_bytes_per_chunk": Threshold of unused memory in an allocated chunk of arena memory after 
+     crossing which the current chunk is chunked into 2.
+  * "initial_regrowth_chunk_size_bytes_after_shrink": Size of first allocation post-shrink. Only relevant 
+    if arena strategy is `kNextPowerOfTwo`. Use -1 to allow ORT to choose the default.
+  */
+  ORT_API2_STATUS(CreateArenaCfgV2, _In_reads_(num_keys) const char* const* arena_config_keys,
+                  _In_reads_(num_keys) const size_t* arena_config_values, _In_ size_t num_keys,
```

Comment:
Can we call this CreateArenaConfig? 

Hopefully this new version will be more permanent and can have a nice name. :P

Reply:
I actally feel it is better to be explicit that this is a V2 version of an API. Otherwise we are going to have CreateArenaCfg and CreateArenaConfig in the same api struct...